src/Data/List/Relation/Unary/Any/Properties.agda

```diff
@@ -255,24 +263,41 @@ Any-×⁻ pq with Product.map₂ (Product.map₂ find) (find pq)
 
     (p , q) ∎
 
-  to∘from : ∀ pq → Any-×⁺ {xs = xs} (Any-×⁻ pq) ≡ pq
-  to∘from pq with find pq
-      | (λ (f : (proj₁ (find pq) ≡_) ⋐ _) → map∘find pq {f})
-  ... | (x , x∈xs , pq′) | lem₁
-    with find pq′
-      | (λ (f : (proj₁ (find pq′) ≡_) ⋐ _) → map∘find pq′ {f})
-  ... | (y , y∈ys , p , q) | lem₂
-    rewrite sym $ map-∘ {R = λ x → Any (λ y → P x × Q y) ys}
-                        (λ p → Any.map (λ q → p , q) (lose y∈ys q))
-                        (λ y → subst P y p)
-                        x∈xs
-            = lem₁ _ helper
-    where
-    helper : Any.map (λ q → p , q) (lose y∈ys q) ≡ pq′
-    helper rewrite sym $ map-∘ (λ q → p , q)
-                               (λ y → subst Q y q)
-                               y∈ys
-           = lem₂ _ refl
+  to∘from : ∀ pq → Any-×⁺ (Any-×⁻ pq) ≡ pq
+  to∘from pq =
+    let x , x∈xs , pq′ = find pq
+
+        lem₁ : {f : (x ≡_) ⋐ λ x → Any (λ y → (P x) × (Q y)) ys} →
+               (f-refl : f refl ≡ pq′) → Any.map f _ ≡ pq
+        lem₁ {f} = map∘find pq {f = f}
+
+        y , y∈ys , p , q = find pq′
+
+        lem₂ : {g : (y ≡_) ⋐ λ y → (P x) × (Q y)} →
+               (g-refl : g refl ≡ (p , q)) → Any.map g _ ≡ pq′
+        lem₂ {g} = map∘find pq′ {f = g}
+
+        helper : Any.map (p ,_) (lose y∈ys q) ≡ pq′
+        helper = begin
+          Any.map (p ,_) (lose y∈ys q)
+            ≡⟨ map-∘ (p ,_) (λ y → resp Q y q) y∈ys ⟨
+          _
+            ≡⟨ lem₂ refl ⟩
+          pq′
+            ∎
+
+        h : P ⋐ λ x → Any (λ y → (P x) × (Q y)) ys
+        h p = Any.map (p ,_) (lose y∈ys q)
+
+    in  begin
+      Any-×⁺ (Any-×⁻ pq)
+        ≡⟨⟩
+      Any.map h (lose x∈xs p)
+        ≡⟨ map-∘ h (λ z → resp P z p) x∈xs ⟨
+      _
```

Comment:
If you're going to be this verbose, then this `_` really ought to be spelled out! (Probably the one above in the helper proof as well)

Reply:
Done, but not obvious this illuminates any better compared to the typechecker being able to infer it via `_`... but part of the verbosity was simply not being able to read, much less understand, the existing proofs...

Reply:
The point is that I, as a reader, couldn't easily reconstruct what the typechecker could, so I didn't quite understand what those 2 steps were doing.